pypeit/scripts/install_ql_masters.py

```diff
@@ -69,13 +69,12 @@ def main(args):
                                         'and/or re-download the zip file.')
 
         # Get the pypeit data directory and make sure it exists
-        data_dir = resource_filename('pypeit', 'data')
-        if not os.path.isdir(data_dir):
-            raise NotADirectoryError(f'Unable to find {data_dir}.  Check your installation.')
+        if not os.path.isdir(data.Paths.data):
+            raise NotADirectoryError(f'Unable to find {data.Paths.data}.  Check your installation.')
```

Comment:
Maybe we should do this kind of checking in the `Paths` class?

Reply:
Moved this checking (for all hardwired paths) into the `Paths` class.  This necessitated the use of a metaclass (for now -- `python>=3.9` will allow for the combination of `@classmethod` with `@property`), so the diff is kinda big.  Essentially, the directory attribute (_e.g._, `data.Paths.telgrid`) calls a `@property` method in the metaclass which returns the directory path if it passes the existance check.